walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,67 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> attendees = request.getAttendees();
+
+    // Find times that don't work
+    ArrayList<TimeRange> badRanges = new ArrayList<>();
+    for (Event event : events) {
+      Collection<String> eventAttendees = event.getAttendees();
+      for (String attendee : attendees) {
+        if (eventAttendees.contains(attendee)) {
```

Comment:
depending on the underlying class implementing eventAttendees, this could be an O(1) lookup (e.g. HashSet<String>) or an O(n) lookup (e.g. ArrayList<String>).

To ensure optimal performance, you need to manually convert either "attendees" or "eventAttendees" into a HashSet. It probably makes more sense to do this for attendees, instead of doing it for every event.